Bare bones administration interface with channel creation.

jonsugar opened the pull request:
This pull request adds an initial bare-bones administration interface and the ability to create channels.

### Main features

- Adds _isAdmin_ column to the user table.
- Adds simple scaffolding for the administration area which can be leveraged in future functionality.
- Protects the administration section from access by a non administrator.
- Adds the ability to create channels via the administration interface.
- Adds a default administrator upon database migration or seeding.

![admin-interface](https://user-images.githubusercontent.com/1974648/34698536-7dd0d64a-f4d0-11e7-95cd-467bf69d15a9.jpeg)

Note: this is my first ever pull request so it may need lots of tweaking or have stuff missing so go easy on me 😄.

jonsugar commented on app/User.php:
```diff
@@ -93,16 +93,6 @@ public function confirm()
-     */
-    public function isAdmin()
-    {
-        return in_array($this->name, ['JohnDoe', 'JaneDoe']);
```
Replaced by isAdmin column on the user table.

MattMangoni commented:
Nah, I think we need a proper way to grant access to the admin panel.
A simple middleware and a basic "roles" table would probably be enough for now.

Or maybe we can use a policy for consistency, since that approach is already been used in other parts of the application :)

jonsugar commented:
@m1guelpf and @MattMangoni I agree with both of your comments. I looked at implementing a policy as well as using the Spatie permissions package, https://github.com/spatie/laravel-permission.

In the end, I decided not do it in this PR as there are a number of ways it can be done and I wasn't sure which way to go. 

Also, as @JeffreyWay said in one of the videos, the forum is in a pre-alpha state and I decided that doing just enough to be able to create channels while providing an initial administration scaffold was the way to go for now. 

Sorry, I should have made my approach clear in the PR description. 

Having said all that I'm happy to add any changes we agree upon. My personal preference would be to merge the current code and then I would start work on the agreed changes and submit a further PR.

m1guelpf commented:
@jonsugar Another good approach would be hardcoding admin emails in the code, something like Laravel Spark does. This will allow to remove a database field and the seeding.

MattMangoni commented:
@jonsugar that's a perfectly reasonable approach, mine was nothing more than a suggestion, you already did a great, useful job. At least we can insert and manage channels without having to tinker with Sequel Pro and, more importantly, without having to clear the cache every five minutes :)

@m1guelpf I have a TODO in my PR about creating a master account as part the installation process.
We just need to ask the user for credentials while still providing some default values.

jonsugar commented:
@m1guelpf your suggestion to follow the Laravel Spark method was a good one. I have implemented this. 

@MattMangoni I have added a `council.php` config file with an `administrators` variable. Maybe your install command could hook into that?

Chinese1904 commented:
FYI, app.css is still missing in this PR

jonsugar commented:
@Chinese1904 I'm not sure I understand. I did not make changes to the css.

Chinese1904 commented:
Actually, when reviewing it, it seems like you pushed the minified public/css/app.css file. As you mentioned, that file has no changes, so it shouldn't be in this PR. 👍

MattMangoni commented:
@Chinese1904 that probably comes from the origin/master merge he did :)

JeffreyWay commented:
Looks good, @jonsugar! Thanks. I'll give it a full review soon and merge it in.

JeffreyWay commented:
Great work, @jonsugar! Merged.

kawagan commented on resources/views/admin/channels/index.blade.php:
```diff
@@ -0,0 +1,31 @@
+                    <td>{{$channel->name}}</td>
+                    <td>{{$channel->slug}}</td>
+                    <td>{{$channel->description}}</td>
+                    <td>{{ count($channel->threads()) }}</td>
```
here, should be {{ count($channel->threads) }}
